### PR TITLE
New CI that performs a basic connectivity test

### PR DESCRIPTION
**Pull request by guusdk**

### CODE REVIEW

**Comment by guusdk** [comment]
The new test fails, as it is exposing a bug in Openfire (OF-2156).

**Review by akrherz — APPROVED** [review]
I defer to Dan on this one.

File: runConnectivityIntegrationTests [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+        final int iterations = 25;
+
+        // Execute system under test.
+        int result = 0;
+        for (int i=0; i<iterations; i++) {
+            final XMPPBOSHConnection con = new XMPPBOSHConnection(config);
+            con.connect();
+            try {
+                con.login("john", "secret");
+                Thread.sleep(200);
```
Comment (Fishbowler) [inline runConnectivityIntegrationTests:174]: This is okay, but if we're seeing it flake in the future (e.g. if the local Openfire isn't performant on the GH CI infrastructure), we could replace this magic number with a listener pattern instead.

**Review by Fishbowler — APPROVED** [review]
I've left one comment for "future us" but LGTM 👍

**Pull request merged**